cli/start.go

```diff
@@ -480,6 +480,39 @@ completed, the server exits.
 	RunE:         runQuit,
 }
 
+// waitForShutdown waits until the gRPC streams is closed by the
+// remote end.
+func waitForShutdown(stream serverpb.Admin_DrainClient) error {
+	for {
+		if _, err := stream.Recv(); !grpcutil.IsClosedConnection(err) {
+			return err
+		}
+		return nil
+	}
+}
+
+// doShutdown attempts to trigger server shutdown. The
+// bool return value indicates whether the state is uncertain
+// and thus whether it makes sense to retry the shutdown.
+func doShutdown(c serverpb.AdminClient, ctx context.Context, onModes []int32) (bool, error) {
+	stream, err := c.Drain(ctx, &serverpb.DrainRequest{
+		On:       onModes,
+		Shutdown: true,
+	})
+	if err != nil {
+		return false, err
+	}
+	_, err = stream.Recv()
+	if err == nil {
+		err = waitForShutdown(stream)
+		if err == nil {
+			fmt.Println("ok")
+		}
+		return false, err
+	}
+	return true, err
```

Comment:
Don't return `err` here. If this method returns an error, that's it. If it doesn't return an error, the caller gets to check the boolean (and maybe retry).